packages/web-components/src/components/tree-view/tree-node.ts

```diff
@@ -0,0 +1,385 @@
+import { LitElement, html } from 'lit';
+import { property } from 'lit/decorators.js';
+import { prefix } from '../../globals/settings';
+import CaretDown16 from '@carbon/icons/lib/caret--down/16.js';
+
+import styles from './tree-view.scss?lit';
+import { carbonElement as customElement } from '../../globals/decorators/carbon-element';
+
+/**
+ * Tree node.
+ *
+ * @element cds-tree-node
+ */
+@customElement(`${prefix}-tree-node`)
+class CDSTreeNode extends LitElement {
+  private _hasChildren = false;
+  private _hasIcon = false;
+  /**
+   * Handles `slotchange` event.
+   */
+  private _handleSlotChange({ target }: Event) {
+    const { depth, disabled } = this;
+    const items = (target as HTMLSlotElement)
+      .assignedNodes()
+      .filter(
+        (node) =>
+          node.nodeType === Node.ELEMENT_NODE &&
+          (node as Element).tagName.toLowerCase() === `${prefix}-tree-node`
+      );
+
+    items.forEach((item) => {
+      (item as CDSTreeNode).depth = depth + 1;
+      (item as CDSTreeNode).disabled = disabled;
+    });
+
+    this._hasChildren = items.length > 0;
+    if (this._hasChildren) this.setAttribute('parent', '');
+    this.requestUpdate();
+  }
+
+  /**
+   * Handles icon's `slotchange` event.
+   */
+  private _handleIconSlotChange({ target }: Event) {
+    this._hasIcon = (target as HTMLSlotElement).assignedNodes().length > 0;
+    if (this._hasIcon) this.setAttribute('has-icon', '');
+    this.requestUpdate();
+  }
+
+  /**
+   * Handles style updates based on depth
+   */
+  private _handleStyles = () => {
+    const { depth, _hasChildren: hasChildren, _hasIcon: hasIcon } = this;
+    const calcOffset = () => {
+      if (hasChildren && hasIcon) {
+        return depth + 1 + depth * 0.5;
+      }
+
+      if (hasChildren) {
+        return depth + 1;
+      }
+
+      if (hasIcon) {
+        return depth + 2 + depth * 0.5;
+      }
+
+      return depth + 2.5;
+    };
+    const label = this.shadowRoot!.querySelector(
+      `.${prefix}--tree-node__label`
+    );
+    if (label) {
+      (label as HTMLElement).style.marginInlineStart = `-${calcOffset()}rem`;
+      (label as HTMLElement).style.paddingInlineStart = `${calcOffset()}rem`;
+    }
+  };
+
+  _handleToggleClick = (event) => {
+    const { disabled, href } = this;
+    if (disabled) return;
+    event.stopPropagation();
+    if (href) {
+      event.preventDefault();
+    }
+    this.isExpanded = !this.isExpanded;
+    if (this.hasAttribute('aria-expanded')) {
+      this.setAttribute('aria-expanded', String(this.isExpanded));
+    }
+  };
+
+  /**
+   * sets if tree node is active
+   */
+  @property({ type: Boolean, reflect: true })
+  active = false;
+
+  /**
+   * disabled property
+   */
+  @property({ type: Boolean, reflect: true })
+  disabled = false;
+
+  /**
+   * **Note:** this is controlled by the parent TreeView component, do not set manually.
+   * TreeNode depth to determine spacing
+   */
+  @property({ reflect: true })
+  depth = 0;
```

Comment:
```suggestion
  @state()
  protected _depth = 0;
```

Given the comments, should this be private?